### ISSUE TITLE
NN with dense layers on the MNIST database give NaN in single precision

### ISSUE THREAD

**Issue by ddante** [ddante/NilDa#18]
The NN give NaN after 1/2 epochs with single precision. This happens when relu and softmax/sigmoid layer because the output goes out of bound. Probably some normalization is required.

See:
https://developer.nvidia.com/blog/mixed-precision-training-deep-neural-networks/
